### ISSUE TITLE
Weird Rendering glitch when applying OSD Overlay.

### ISSUE THREAD

**Issue by keggyhander** [fpv-wtf/wtfos-configurator#289]
I thought there was something wrong with my Caddx Vista when I was getting weird glitches in my videos. It was only when I compared the raw video with the ones that had had the OSD rendered in that I noticed the source of the glitch was your software process. Here's a video showing the glitch: https://youtu.be/Oum5dfal44M

Here's a video in split screen showing raw video against the OSD-overlaid version:
https://www.youtube.com/watch?v=z6U-QKHOX0k

I would like to solve this because I like the look of my particular OSD

Cheers

**Comment by keggyhander** [comment]
https://mega.nz/folder/QPJ0TB6B
key:DGpuaAjlUxuXYCDFkCdTwA

**Comment by FinalFrag** [comment]
The output video looks good when I run it through the osd-overlay that is currently on the develop branch. Could you try it again? If the issue is still there, what machine are you running it on? PC/Mac/MacM1?

**Comment by keggyhander** [comment]
PC win 10 64

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by keggyhander** [comment]
Tried it again from here: https://knifa-develop.fpv.wtf/osd-overlay
If anything it was even worse than before. 
What is this develop branch you speak of?

**Comment by FinalFrag** [comment]
I forgot the current development branch has not been released publicly yet. Nevermind that.

I ran your files through that version as well, and they come out ok. Must be something specific about your setup. Could you try another browser and see if that has any effect? What browser did you use now?

**Comment by stylesuxx** [comment]
The develop branch is actually being automatically deployed to https://develop.fpv.wtf/

**Comment by keggyhander** [comment]
Losing patience with this now. If it doesn't work in chrome, what use is it, really?

**Comment by stylesuxx** [comment]
Alright, if you don't want help and don't have time to invest, feel free not to use it.

**Comment by keggyhander** [comment]
I spent plenty of time and energy rooting my devices in order to use it and spent ages setting up my optimum OSD set up. I don't quite know what more you expect me to do on my end to remove this annoying glitch in the overlay render.

________________________________

**Comment by keggyhander** [comment]
I don't have any other browsers installed. Which one would YOU suggest?
________________________________

**Comment by keggyhander** [comment]
Version 110.0.5481.178 (Official Build) (64-bit)

I'm not being entitled at all. You're the clever ones. I would expect you to have more answers than me. It creates the file, just with weird glitches. I bet there are more complaints from people for whom the file doesn't even start rendering. My impatience is whether to bother with it at all. Not with you. Have you ever seen this glitch before? If you haven't, then just say so.
________________________________

**Comment by keggyhander** [comment]
Right, I've downloaded Edge, and tried it on there. It still presents a very minor glitch in one place, but is very much improved. With that in mind, what do suggest is at the heart of the problem?
________________________________

**Comment by FinalFrag** [comment]
I'm running the same version of chrome as you do. I also tried it in edge. Both result in no glitches.

At this point, I have no idea what is at the heart of this problem.

I have seen glitching before in my own files that were edited, but never in files straight from the goggles. I analysed your file and it has all the correct info in it that it should process fine.

I need to be able to reproduce the issue on my own machine before I can start digging. I'm gonna try to set up a clean install of windows in a virtual machine tomorrow, see if that makes a difference.

When you re-run the files in the same browser, do the glitches always appear in the same spots or is it random?

**Comment by keggyhander** [comment]
Virtually/exactly the same.
________________________________

**Comment by j005u** [comment]
This sounds very likely related to some combination of Chrome and your graphics cards drivers or something along those lines messing up. The support is relatively new and not that robust yet on all devices.

Updating GPU drivers could maybe possibly help. @Knifa is there an easy way to force CPU encoding, to give that try instead? Presuming Chrome defaults to GPU.

PS: Next time, if you want help, please refrain from using statements like "loosing patience" indeed. It works for 99%+ of users. Chrome WebCodecs do indeed have weird edge case bugs depending on the rest of your software/hardware, which are entirely Google's bugs. We do our best to work through them as they come up and solve them if we can. When we have the spare time to do so, completely for free.

So please consider what amount of patience and effort it takes from the devs before taking out your frustration here.

Hope we're able to get you sorted in the end.

**Comment by preisedj** [comment]
Same here.  https://youtu.be/pXApHgllCmA

**Comment by FinalFrag** [comment]
> Same here. https://youtu.be/pXApHgllCmA

Could you upload the source files somewhere?

**Comment by preisedj** [comment]
Here are the files I'm working with.  My flight controller is using Ardupilot, but I have the same results using INav.  

https://drive.google.com/drive/folders/1hoAYZ6wh9drOr3gKww1oy9TIsraoYH_x?usp=sharing

**Comment by j005u** [comment]
@preisedj the fonts you provided on google drive are Betaflight fonts. Please see the issue you opened for the correct font name pattern you should be looking for.
